src/SqlBindingUtilities.cs

```diff
@@ -134,30 +134,11 @@ public static SqlCommand BuildCommand(SqlAttribute attribute, SqlConnection conn
         /// <summary>
         /// Returns a dictionary where each key is a column name and each value is the SQL row's value for that column
         /// </summary>
-        /// <param name="reader">
-        /// Used to determine the columns of the table as well as the next SQL row to process
-        /// </param>
-        /// <param name="cols">
-        /// Filled with the columns of the table if empty, otherwise assumed to be populated
-        /// with their names already (used for cacheing so we don't retrieve the column names every time)
-        /// </param>
+        /// <param name="reader">Used to determine the columns of the table as well as the next SQL row to process</param>
         /// <returns>The built dictionary</returns>
-        public static Dictionary<string, string> BuildDictionaryFromSqlRow(SqlDataReader reader, List<string> cols)
+        public static IReadOnlyDictionary<string, string> BuildDictionaryFromSqlRow(SqlDataReader reader)
```

Comment:
Is this something directly related to your PR? In general this kind of cleanup was something I was hoping we'd get around to eventually so I'm not opposed to looking at doing this, but it's general enough that I'd rather it be split into its own PR.

Reply:
The change to method `BuildDictionaryFromSqlRow` simplifies the way it gets called from `SqlTableChangeMonitor.cs`. I am okay with moving this change to a separate PR if that works.

Reply:
@JatinSanghvi Please merge in latest from main so these changes don't show up.